tasks/pre_checks/002_validate_hostname_tasks.yml

```diff
@@ -37,21 +37,45 @@
     fail:
       msg: >
         The address proposed for this host does not resolves locally
-    when: r_host_address not in ansible_all_ipv4_addresses
+    when: r_host_address not in ansible_all_ipv4_addresses | union(ansible_all_ipv6_addresses)
   - name: Ensure the resolved address resolves on the selected interface
     block:
-    - name: Get target address from selected interface
+    - name: Get target address from selected interface (IPv4)
       set_fact:
-        target_address: "{{ hostvars[inventory_hostname]['ansible_' + he_mgmt_network.replace('-','_')]['ipv4']['address'] if 'ansible_' + he_mgmt_network.replace('-','_') in hostvars[inventory_hostname] else hostvars[inventory_hostname]['ansible_' + he_bridge_if.replace('-','_')]['ipv4']['address'] }}"
-    - debug: var=target_address
+        target_address_v4: "{{
+          hostvars[inventory_hostname]['ansible_' + he_mgmt_network.replace('-','_')]['ipv4']['address']
+            if 'ansible_' + he_mgmt_network.replace('-','_') in hostvars[inventory_hostname] and
+              'ipv4' in hostvars[inventory_hostname]['ansible_' + he_mgmt_network.replace('-','_')]
+          else
+          hostvars[inventory_hostname]['ansible_' + he_bridge_if.replace('-','_')]['ipv4']['address']
+            if 'ansible_' + he_bridge_if.replace('-','_') in hostvars[inventory_hostname] and
+              'ipv4' in hostvars[inventory_hostname]['ansible_' + he_bridge_if.replace('-','_')]
+          else
+          None
+        }}"
+    - debug: var=target_address_v4
+    - name: Get target address from selected interface (IPv6)
+      set_fact:
+        target_address_v6_list: "{{
+          hostvars[inventory_hostname]['ansible_' + he_mgmt_network.replace('-','_')]['ipv6'] | map(attribute='address') | list
+            if 'ansible_' + he_mgmt_network.replace('-','_') in hostvars[inventory_hostname] and
+              'ipv6' in hostvars[inventory_hostname]['ansible_' + he_mgmt_network.replace('-','_')]
+          else
+          hostvars[inventory_hostname]['ansible_' + he_bridge_if.replace('-','_')]['ipv6'] | map(attribute='address') | list
+            if 'ansible_' + he_bridge_if.replace('-','_') in hostvars[inventory_hostname] and
+              'ipv6' in hostvars[inventory_hostname]['ansible_' + he_bridge_if.replace('-','_')]
+          else
+          []
```

Comment:
Should be `None` like it is in the IPv4 section on line 54

Reply:
No, unfortunately on IPv4 it's a scalar while on IPv6 we need a list, maybe empty, but a list